tests/constraints/tests.py

```diff
@@ -80,21 +83,38 @@ def test_repr(self):
     def test_deconstruction(self):
         fields = ['foo', 'bar']
         name = 'unique_fields'
-        check = models.UniqueConstraint(fields=fields, name=name)
-        path, args, kwargs = check.deconstruct()
+        constraint = models.UniqueConstraint(fields=fields, name=name)
+        path, args, kwargs = constraint.deconstruct()
         self.assertEqual(path, 'django.db.models.UniqueConstraint')
         self.assertEqual(args, ())
         self.assertEqual(kwargs, {'fields': tuple(fields), 'name': name})
 
+    def test_deconstruction_with_condition(self):
+        fields = ['foo', 'bar']
+        name = 'unique_fields'
+        constraint = models.UniqueConstraint(fields=fields, name=name, condition=models.Q(foo=models.F('bar')))
+        path, args, kwargs = constraint.deconstruct()
+        self.assertEqual(path, 'django.db.models.UniqueConstraint')
+        self.assertEqual(args, ())
+        self.assertEqual(kwargs, {'fields': tuple(fields), 'name': name, 'condition': models.Q(foo=models.F('bar'))})
+
     def test_database_constraint(self):
         with self.assertRaises(IntegrityError):
-            Product.objects.create(name=self.p1.name)
+            Product.objects.create(name=self.p1.name, color=self.p1.color)
 
     def test_model_validation(self):
-        with self.assertRaisesMessage(ValidationError, 'Product with this Name already exists.'):
-            Product(name=self.p1.name).validate_unique()
+        with self.assertRaisesMessage(ValidationError, 'Product with this Name and Color already exists.'):
+            Product(name=self.p1.name, color=self.p1.color).validate_unique()
+
+    def test_model_validation_with_condition(self):
+        Product(name=self.p1.name, color='blue').validate_unique()
+        Product(name=self.p2.name).validate_unique()  # validation ignored, will raise on DB level
```

Comment:
I think comment would be better as a docstring:
`"""Partial unique constraints are ignored by Model.validate_unique()."""`

Reply:
changed